altair/utils/core.py

```diff
@@ -200,7 +201,12 @@ def infer_dtype(value):
 ]
 
 
-def infer_vegalite_type(data):
+InferredVegaLiteType = Literal["ordinal", "nominal", "quantitative", "temporal"]
```

Comment:
I think it can also be a `geojson` type here? See https://altair-viz.github.io/user_guide/generated/core/altair.Type.html for the options. For my own understanding, can the available type options also be inferred from the schema here? And if so, is there a reason not to use this? Eg. if type `array` is added to Vega-Lite (not sure if this happens) then type support for this requires a manual update here?

Reply:
I think I see where you're coming from. However, this type is used to annotate what values the `infer_vegalite_type` function can return. It is not used in the sense of a schema for all possible vega lite types. It should therefore only contain the types that can be returned and this would only change if we manually update that function. Currently, the function cannot return `geojson`.

Keeping the concepts of types and VL schema separate might be challenging during this whole process. I also struggle with it and I agree that where possible we should infer one from the other.

As a side note: In the last commit I renamed `InferredVegaLiteType` to `_InferredVegaLiteType` so it's private which is in line with the discussion we had [here](https://github.com/altair-viz/altair/pull/3081#issuecomment-1587799122).

Reply:
Thanks for commenting. That clarifies it a bit more!
Not sure yet about the public names I'm not sure we should place words like `Return` and `Func` in type definitions.